### DISCUSSION TITLE
API Route Timeout

### DISCUSSION THREAD

**Discussion by employee451** [vercel/next.js#23011] [Help]
Hi there.
I have an API route that runs once every month. It is very heavy, meaning it has to do a lot calculations and send emails, etc. Due to this, it also takes quite a long time to run.
My issue is that it sometimes results in a timeout before it gets to the end and sends a `res.status(200)` back.
Is it recommendable to just at the very beginning send back `res.status(200)` (without returning), and then do all of the calculations? Or how should this be done?
Thanks in advance

**Comment by ErnestBrandi** [comment]
Does this happens in production ? Where is the website hosted ?

**Reply by employee451** [reply]
Yes, this happens in production on Vercel

**Reply by ErnestBrandi** [reply]
https://vercel.com/docs/platform/limits#serverless-function-execution-timeout

**Reply by employee451** [reply]
Thanks for that – I should have that up myself. Do you know if I can send the `res.status(200)` at the beginning of the function? It is recommended by the cron-job service I use to immediately send back a 200 response before doing the heavy lifting.

**Reply by ErnestBrandi** [reply]
Setting the status at the beginning of your function doesn't make any sens. Plus setting the status does nothing special by itself, it is usually followed by a `json`, `send` or `end` function. You cannot avoid the Vercel limitation on functions execution.
